src/main/java/domain/Match.java

```diff
@@ -0,0 +1,77 @@
+package domain;
+
+public enum Match {
+    THREE(3, 5000) {
+        @Override
+        public String print() {
+            return getCount() + "개 일치 (" + getPrice() + "원)- ";
+        }
+
+    },
+    FOUR(4, 50000) {
+        @Override
+        public String print() {
+            return getCount() + "개 일치 (" + getPrice() + "원)- ";
+        }
+    },
+    FIVE(5,1500000) {
+        @Override
+        public String print() {
+            return getCount() + "개 일치 (" + getPrice() + "원)- ";
+        }
+    },
+    FIVEWITHBONUS(5,30000000) {
+        @Override
+        public String print() {
+            return getCount() + "개 일치, 보너스 볼 일치(" + getPrice() + "원) - ";
+        }
+    },
+    SIX(6, 2000000000) {
+        @Override
+        public String print() {
+            return getCount() + "개 일치 (" + getPrice() + "원)- ";
+        }
+    };
+
+    private final int price;
+    private final int count;
+
+    Match(int count, int price) {
+        this.count = count;
+        this.price = price;
+    }
+
+    public int getPrice() {
+        return price;
+    }
+
+    public int getCount() {
+        return count;
+    }
+
+    public abstract String print();
+
+    public static Match from(int c, boolean bonus) {
+        if (c == THREE.count) {
+            return THREE;
+        } else if (c == FOUR.count) {
+            return FOUR;
+        } else if (c == FIVE.count && !bonus) {
+            return FIVE;
+        } else if (c == FIVE.count && bonus) {
+            return FIVEWITHBONUS;
+        } else if (c == SIX.count) {
+            return SIX;
+        }
```

Comment:
Enum 객체의 경우 이처럼 열거된 값에 맞게 객체가 생성되어야 하는 상황이 많이 발생해요.
이런 패턴도 있답니다!
```java
public static Match from(int c, boolean bonus) {
    Arrays.stream(Match.values())
            .filter(match -> match.count == c)
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("해당하는 Match가 없습니다."));
}
```
요런 방법도 활용 해보시고 bonus 처리도 적용하면 좋을 것 같아요! (모두 stream 내에서 이루어질 필요는 없어요)

그리고, early return을 했으니 else if는 필요 없을 것 같아요.
```suggestion
        if (c == THREE.count) {
            return THREE;
        } 
        if (c == FOUR.count) {
            return FOUR;
        } 
        if (c == FIVE.count && !bonus) {
            return FIVE;
        } 
        if (c == FIVE.count && bonus) {
            return FIVEWITHBONUS;
        } 
        if (c == SIX.count) {
            return SIX;
        }
```